doc/stdenv/stdenv.chapter.md

```diff
@@ -317,6 +317,61 @@ The script will be usually run from the root of the Nixpkgs repository but you s
 
 For information about how to run the updates, execute `nix-shell maintainers/scripts/update.nix`.
 
+### Recursive attributes in `mkDerivation`
+
+If you pass a function to `mkDerivation`, it will receive as its argument the final arguments, considering use of `overrideAttrs`. For example:
+
+```nix
+mkDerivation (finalAttrs: {
```

Comment:
`finalAttrs` is unambiguous, avoiding name shadowing when used in an overlay that uses names conforming to what's expected from flakes - `final` and `prev` are taken.

`self` is from the older overlay naming convention, which is not great but less of an issue for name shadowing reasons. I would like to include `attrs` though, because it conforms with `overrideAttrs`, hinting a tiny bit that it's about the `mkDerivation` arguments, not the result.